doc/manual/syntax.md

```diff
@@ -0,0 +1,446 @@
+The fields are strings, but can be noted as identifiers if they respect identifiers syntax (internally, they will be stored as strings). Values can be of any type.
+Elements inside a record are unordered.
+Two records can be _merged_ together using the operator `&`. The reader can find more information about merging in the relevant documentation.
+
+Examples:
+```
+{}
+{a = 3}
+{my_id_n5 = "my id number 5", my_id_n4 = }
+{"5" = 5, six = 6}
+```
+
+Accessing a record field can be done using the `.` operator :
+```
+> { a = 1, b = 5 }.a
+1
+
+> { a = 1 }.b
+error: Missing field
+
+> { "1" = "one" }."1"
+"one"
+```
+
+It is possible to write records of records via the *piecewise syntax*, where we separate fields by dots:
+```
+> { a = { b = 1 } }
+{ a = { b = 1 } }
+ 
+> { a.b = 1 }
+{ a = { b = 1 } }
+
+> { a.b = 1, a.c = 2, b = 3}
+{ a = { b = 1, c = 2 }, b = 3 }
+```
+
+Since all fields are strings, we can use string interpolation to create keys or accessing:
+```
+> let k = "a" in { "#{k}" = 1 }
+{ a = 1 }
+
+> let k = "a" in { a = 1 }."#{k}"
+1
+```
+
+## Constructs
+
+### If-Then-Else
+This construct allows conditional branching in your code. You can use it as `if <bool expr> then <expr> else <expr>`.
+
+Examples:
+```
+> if true then "TRUE :)" else "false :("
+"TRUE :)"
+
+> if false then "Not this one" else "This one"
+"This one"
+
+> if "forty-two" == 42 then "equal?" else "unequal"
+"unequal"
+
+> ["1"] @ (if 42 == "42" then ["3"] else ["2"]) @ ["3"]
+["1", "2", "3"]
+```
+
+The `then` and `else` branches are only evaluated accordingly to the result of the expression inside the `if`.
+
+Examples:
+```
+> ["1"] @ (if true then ["2"] else "error") @ ["3"]    // Here only the expression in 'then' is evaluated, which is correct
+["1", "2", "3"]
+
+> ["1"] @ (if false then ["2"] else "error") @ ["3"]    // Here only the expression in 'else' is evaluated, which results in a type error
+error: Type error
+```
+
+### Let-In
+Let-in allows the binding of an expression. It is used as `let <ident> = <expr> in <expr>`.
+
+Examples:
+```
+> let r = { a = "a", b = "b" } in r.a
+"a"
+
+> let inner = { inside = true } in let outer = { outside = inner.inside } in outer.outside
+true
+
+> let a = 1 in let b = 2 in a + b
+3
+```
+
+## Functions
+A function is declared using the `fun` keyword, then arguments separated by spaces, and finally an arrow `=>` to add the body of the function.
+To call a function, just add the arguments after it separated with spaces.
+Functions in Nickel are curried, meaning that a function taking multiple arguments is actually a function that takes a single argument and returns a function taking the rest of the arguments, until it is applied.
+
+Examples:
+```
+> (fun a b => a + b) 1 2
+3
+
+let add = fun a b => a + b in add 1 2
+3
+
+> let add = fun a b => a + b in
+    let add1 = add 1 in
+        add1 2
+3
+```
+
+All existing infix operators in Nickel can be turned into prefix functions by putting them inside parentheses.
```

Comment:
```suggestion
All existing infix operators in Nickel can be turned into functions by putting them inside parentheses.
```
I see what you mean, but to someone especially into parsing or PL (which Nickel users may be), it may sound a bit strange, as if we now have different types of functions, prefix functions, and other. At this point, I think it sufficient and clearer to just say this turns operators into functions.